### PR TITLE
gccxml requires gcc@4.9 at runtime

### PR DESCRIPTION
**Pull request by muffgaga**
* I did not test if other old compilers are supported.
* Newer `gcc`s introduce syntax in header which cannot
  be parsed by gccxml.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gccxml/package.py [hunk 1]
```diff
@@ -36,3 +37,17 @@ class Gccxml(CMakePackage):
     patch('darwin-gcc.patch', when='%gcc platform=darwin')
     # taken from https://github.com/gccxml/gccxml/issues/11#issuecomment-140334118
     patch('gcc-5.patch', when='%gcc@5:')
+
+    depends_on('gcc@:4.9.999', type='run')
+
+    @run_after('install')
+    def post_install(self):
```
Comment (healther) [inline var/spack/repos/builtin/packages/gccxml/package.py]: rename function to indicate what it does. E.g. patch_runtime_compiler_path (if that is what it is doing)

Comment (muffgaga) [reply var/spack/repos/builtin/packages/gccxml/package.py]: done

**Comment by healther** [comment]
@adamjstewart can you merge this?

**Comment by alalazo** [comment]
Closing, since the PR is really old and GCCXML has been deprecated in favor of https://github.com/CastXML/CastXML Feel free to reopen if you want to push this forward.

**Pull request closed without merging**